risc0/zkvm/src/host/client/env.rs

```diff
@@ -388,4 +399,11 @@ impl<'a> ExecutorEnvBuilder<'a> {
         self.inner.input_digest = Some(digest);
         self
     }
+
+    /// Add a callback for coprocessor requests.
+    #[stability::unstable]
+    pub fn coprocessor_callback(&mut self, handler: CoprocessorCallbackRef<'a>) -> &mut Self {
```

Comment:
Could this be `dyn CoprocessorCallback + 'a` or `impl CoprocessorCallback + 'a`, similar to how we use `impl Write + 'a` above?

Reply:
It could, but I basically need to make a copy of the ref anyways, so immediately I need to throw it into a Rc/RefCell. I'm finding from an API perspective it's just easier for both caller and callee to assume Rc/RefCell.